### PR TITLE
Sanitize effector parameters when reporting an exception

### PR DESCRIPTION
**Pull request by bostko**
It is possible to show such parameters on other places.

### CODE REVIEW

**Comment by ygy** [comment]
LGTM

**Comment by aledsage** [comment]
Whoever merges this, please look at marking https://issues.apache.org/jira/browse/BROOKLYN-282 as resolved.

**Comment by aledsage** [comment]
@bostko I've created https://github.com/bostko/brooklyn-server/pull/1 - please review. If you merge it into your branch, then it should automatically appear in this PR.

For the apache record, that PR does:

    Also refactors EffectorUtils.prepareArgsForEffector (deprecating old
    method), and updates how MethodEffector calls it so we have the 
    parameter names.

    The previous calls to `sanitizeArgs(args)` where args was not of type `Map` would do nothing. 
    Therefore if this was ever called with an array etc, we'd log all of the parameter values.

File: core/src/test/java/org/apache/brooklyn/core/mgmt/internal/TestEntityWithEffectors.java [hunk 1]
```diff
@@ -21,15 +21,77 @@
...
+ * export JAVA_OPTS="-Xms256m -Xmx1g -XX:MaxPermSize=256m -Dlogback.configurationFile=/path/to/logback-trace.xml"
+ * ./bin/brooklyn launch --persist auto --persistenceDir /path/to/persistedState
+ * }
+ * </pre>
+ * 
+ * Deploy the blueprint below:
+ * <pre>
+ * {@code
+ * services:
+ * - type: org.apache.brooklyn.core.mgmt.internal.TestEntityWithEffectors
```
Comment (bostko) [inline core/src/test/java/org/apache/brooklyn/core/mgmt/internal/TestEntityWithEffectors.java]: I had to add 
```xml
    <dependency>
        <groupId>org.apache.brooklyn</groupId>
        <artifactId>brooklyn-core</artifactId>
        <version>${project.version}</version>
        <classifier>tests</classifier>
    </dependency>
```
In my maven file to use this entity.
@aledsage do you think we should this note here?

Comment (aledsage) [inline core/src/test/java/org/apache/brooklyn/core/mgmt/internal/TestEntityWithEffectors.java]: The way I built it was to export TestEntityWithEffectors*.java as a jar from my IDE, and add it to ./lib/patch of a pre-installed Brooklyn. That didn't require adding the pom.

Which pom did you add that to? Maybe we should add something more general like "You'll have to ensure this is on the classpath of your brooklyn server, before testing". I'll update my PR accordingly.

**Comment by bostko** [comment]
@aledsage thank you for the changes!

**Comment by aledsage** [comment]
Unfortunately jenkins is not running PR builds. I ran `mvn clean install` manually, and it worked. So merging.

**Pull request merged**